lib/jerboa/format/body/attribute/nonce.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Jerboa.Format.Body.Attribute.Nonce do
+  @moduledoc """
+  NONCE attribute as defined in [STUN RFC](https://tools.ietf.org/html/rfc5389#section-15.8)
+  """
+
+  alias Jerboa.Format.Body.Attribute.{Decoder,Encoder}
+  alias Jerboa.Format.Nonce.LengthError
+  alias Jerboa.Params
+
+  defstruct value: ""
+
+  @max_chars 128
+
+  @typedoc """
+  Represent nonce's value
+  """
+  @type t :: %__MODULE__{
+    value: String.t
+  }
+
+  defimpl Encoder do
+    alias Jerboa.Format.Body.Attribute.Nonce
+    @type_code 0x0015
+
+    @spec type_code(Data.t) :: integer
+    def type_code(_), do: @type_code
+
+    @spec encode(Nonce.t, Params.t) :: binary
+    def encode(attr, _), do: Nonce.encode(attr)
+  end
+
+  defimpl Decoder do
+    alias Jerboa.Format.Body.Attribute.Nonce
+
+    @spec decode(Nonce.t, value :: binary, params :: Params.t)
+      :: {:ok, Nonce.t} | {:error, struct}
+    def decode(_, value, _), do: Nonce.decode(value)
+  end
+
+  @doc false
+  def encode(%__MODULE__{value: value}) do
+    if String.valid?(value) && String.length(value) <= @max_chars do
+      value
+    else
+      raise FunctionClauseError
```

Comment:
That's true. I thought that it would be consitent with other `encode/1` functions, which in case of wrong argument just raise FunctionClause (because we use guards there and there is simply no clause matching). Now I think that they all should raise ArgumentError. Is that okay?